### ISSUE TITLE
Feature Suggestions

### ISSUE THREAD

**Issue by Skyyblaze** [torchgm/RoundedTB#2]
This is a wonderful tool, I already added it to my Autostart and don't want to miss it anymore, great job on it!

I do have two feature suggestions though:

- Add a option to hide the tray-icon
- Add a option to add a outline and/or shadow to the taskbar.

Keep up the good work! :)

**Comment by torchgm** [comment]
Hi there! Thanks for the kind words, and glad you're enjoying it.

Regarding the tray icon, if you want to hide it then you're welcome to drag it into the extended system tray and it should automatically dismiss itself there in the future:
![image](https://user-images.githubusercontent.com/31840547/120250095-d4f8c480-c274-11eb-9c98-8c0f18498dc5.png)
If you mean hide it entirely however, that certainly is possible but would mean there would be no way to interact with the application after it's started (and therefore no way to change settings or cleanly close it). However, if this is what you mean then I can probably add a command-line flag or hidden setting that starts it without the tray icon.

As for outlines and shadows, I won't spoil anything but I do have some plans for taskbar effects, so just have this teaser for now: ![image](https://torch.is/sending-you-a/IdolizedFearfulMussaurus.gif)

**Comment by Longshoez** [comment]
Hey!!! so nice to find something like this would it be possible to remove all the space that isn't used and have it automatically resize when opening apps that weren't pinned?.
A few months ago I was trying to develop something similar but I just lost interest in the project, my approach was to make a new task bar from scratch which obviously wasn't the easiest and smartest way to start the project haha.

**Comment by torchgm** [comment]
@Longshoez That's certainly something that can be at least partly implemented; it's probably not going to happen straight away due to how complicated it is though. In doing so you'd lose the entire tray area, unless you broke that out into its own window (which comes with a whole slew of other issues that lead to an incredibly unstable shell).

The premise of just cutting the taskbar down to size isn't very difficult at all, it's keeping track of the number of taskbar icons, finding some way to "reveal" the full taskbar to access the tray, dealing with all sorts of weird quirks such as the taskbar icons resizing themselves based on slate configuration/taskbar fullness, and then managing all this on top of the fact that the taskbar can be moved & resized pretty freely by the user at any time.

In summary, perhaps, but not straight away.

**Comment by MikeCoder96** [comment]
Hello there, i really appreciate this work!
I'm usually keep my taskbar hidden until i press key start but i not see this feature in the code so i started work on it to add a CheckBox to keep taskbar hidden.
Can i make a pull request at the end of implementation of this feature?

**Comment by torchgm** [comment]
> Hello there, i really appreciate this work!
> I'm usually keep my taskbar hidden until i press key start but i not see this feature in the code so i started work on it to add a CheckBox to keep taskbar hidden.
> Can i make a pull request at the end of implementation of this feature?

I'm not 100% sure what you mean; if you're referring to the taskbar auto-hiding when not in use, then yes RoundedTB is preliminarily compatible. Just enable auto-hiding from the Settings app. If this isn't what you mean, then feel free to submit your PR and we can look at it there.

I've updated the issue title to avoid confusion with people suggesting new features here instead of in a new issue.

**Comment by MikeCoder96** [comment]
I mean, totally hide! With auto hide when you go with pointer to the TB it appear and I don't like it so I prefer to show TB only when I press start button.

**Comment by carlosvsilva** [comment]
> Yeah I know that hiding the tray-icon entirely would make access difficult but RoundedTB is an app that I would configure once, set to auto-start and never mess with again, that's why I disable the tray-icons of all customization apps whenever possible so they don't clutter the extended tray area. To access the settings again if the need arises I'm using to starting the app again from the Start Menu while it's already running

I came here to also request this, exactly like Skyyblaze requested. I think that method makes perfect sense: only display the configuration window when a second instance of the app is started. if it's the first, just do its magic and rest in the background without filling up the tray icon list. thanks for the great software!
